src/integrations/admin/addon-installation/installation-integration.php

```diff
@@ -141,13 +150,15 @@ public function activate_addon( $addon_slug ) {
 			// Translators: %s expands to the name of the addon.
 			$output[] = __( 'Addon activated.', 'wordpress-seo' );
 		} catch ( User_Cannot_Activate_Plugins_Exception $exception ) {
-			$output[] = __( 'You are not allowed to activate plugins.', 'wordpress-seo' );
+			$output[]         = __( 'You are not allowed to activate plugins.', 'wordpress-seo' );
+			$addons_activated = false;
```

Comment:
This is using a local variable, not the introduced `$this->addons_activated` class property.